### PR TITLE
Remove `QTensor`

### PR DESCRIPTION
**Pull request by PierreGuilmin**
As discussed together, we don't need it for now. I will build on this PR for incoming small PRs. I placed `to_torch` in the `types` file because I have some other types conversion methods incoming.

### CODE REVIEW

File: torchqdynamics/utils.py [hunk 1]
```diff
@@ -126,3 +129,28 @@ def expect(operator: torch.Tensor, state: torch.Tensor) -> torch.Tensor:
     # TODO: Once QTensor is implemented, check if state is a density matrix or ket.
     # For now, we assume it is a density matrix.
     return torch.einsum('ij,...ji', operator, state)
+
+
+def from_qutip(x: Qobj) -> torch.Tensor:
```
Comment (gautierronan) [inline torchqdynamics/utils.py:134]: Same comment here. It's just a suggestion however.

File: torchqdynamics/types.py [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+        PyTorch tensor.
+    """
+    if isinstance(x, Qobj):
+        return from_qutip(x)
+    elif isinstance(x, get_args(TensorLike)):
+        return torch.as_tensor(x)
+    else:
+        raise TypeError(
+            'invalid type (supported: QuTiP quantum object, NumPy array, '
+            'Python list or PyTorch tensor)'
```
Comment (gautierronan) [inline torchqdynamics/types.py:35]: The error message should rather be something like
```python
f'Input of type {type(x)} is not supported. to_torch only supports blablabla'
```
This is more explicit and the user knows what he did wrong without having to run an additional test.

**Review by gautierronan — CHANGES_REQUESTED** [review]

File: torchqdynamics/types.py [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
 TimeDependentOperator = Union[torch.Tensor, Callable[[float], torch.Tensor]]
 
 # type for objects convertible to a torch tensor using `torch.as_tensor`
 TensorLike = Union[List, np.ndarray, torch.Tensor]
+
+# type for objects convertible to a torch tensor using `torchqdynamics.to_tensor`
+QTensorLike = Union[TensorLike, Qobj]
+
+
+def to_torch(x: QTensorLike) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (PierreGuilmin) [reply torchqdynamics/types.py:19]: I like the `to_torch(A).to(X)` which separates the steps, but I'm open to change to accept `dtype`, `device` and `X`. I propose to wait and see if we use this method a lot or not, and whether the proposed change would be better with our use cases. 😉

[the remark this answers is not in the document]
Comment (PierreGuilmin) [reply torchqdynamics/types.py:19]: I wrote the comment down in a backlog, so we don't loose the idea.

**Review by gautierronan — APPROVED** [review]

**Pull request merged**